app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,68 @@
+class PostsController < ApplicationController
+  before_action :checkSession, only: [:new, :create, :edit, :destroy, :update]
+
+  def destroy
+    @post = Post.find(params[:id])
+    if check_owner
+      @post.destroy
+      flash[:notice] = "Your post has been deleted successfuly"
+      redirect_to root_path
+    end
+  end
+
+  def update
+    if Post.find(params[:id]).update(post_params)
+      flash[:notice] = "Your post has been edited successfuly"
+      redirect_to root_path
+    else
+      flash[:warning] = "There has been a problem editing your post, please try again later"
+      redirect_to root_path
+    end
+  end
+
+  def edit
+    @post = Post.find(params[:id])
+    check_owner
+  end
+
+  def index
+    @posts = Post.all
+  end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
```

Comment:
Remember one thing. We are not writing code only for us. You have to write codes so that others can read those and understand those. In this method, we can do this

```ruby
def create
    @post = current_user.posts.build(post_params)
    if @post.save
      flash[:notice] = "Post Successful"
      redirect_to root_path
    else
      flash.now[:warning] = "Post not saved, something went wrong"
      render "new"
    end
  end
```
This is mostly done by most of the devs. Now to implement these things you have to add an association in the user model. Also that `post_params` method is so confusing. You have to make this as simple as possible. Check below 👇 for `post_params` method.